### PR TITLE
Update the compatibility matrix for br

### PR DESCRIPTION
**Pull request by IANTHEREAL**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Update the compatibility matrix for br. 

Update content from https://github.com/pingcap/docs-cn/pull/10429

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by shichun-0415** [comment]
@IANTHEREAL Please invite a tech reviewer. Thanks.

File: br/br-usage-restore.md [hunk 1]
```diff
@@ -57,7 +57,8 @@ Full Restore <---------/...............................................> 17.12%.
 
 > **注意：**
 >
-> 自 BR v5.1.0 开始，全量备份会备份**系统表数据**。BR v6.2.0 以前的版本，在默认设置下只会恢复用户数据，而不会恢复系统表数据。自 BR v6.2.0 开始，如果备份数据中包含系统表数据，在默认设置下（即没有 `filter` 等参数影响数据恢复范围），恢复数据时 BR 将同时恢复**系统权限相关数据**。详细内容请查看[恢复 `mysql` 数据库下的表](#恢复-mysql-数据库下的表)
+> 自 BR v5.1.0 开始，全量备份会备份**系统表数据**。BR v6.2.0 以前的版本，在默认设置下恢复只会恢复用户数据，而不会恢复系统表数据。自 BR v6.2.0 开始，在默认设置下（即没有 `filter` 等参数影响数据恢复范围），恢复数据时 BR 将同时恢复**系统权限相关数据**。该功能也带来了一些兼容性改变，同样在默认配置下，`br restore full` 不再支持恢复数据到已有数据的集群，否则需要过滤掉相关的系统表才能恢复，参考命令 `br restore full --filter '.' --filter '!mysql.*'`
```
Comment (ran-huang) [inline br/br-usage-restore.md:60]:
```suggestion
> 自 BR v5.1.0 开始，全量备份会备份**系统表数据**。BR v6.2.0 以前的版本，在默认设置下恢复只会恢复用户数据，而不会恢复系统表数据。
> 
> 自 BR v6.2.0 开始，在默认设置下（即没有 `filter` 等参数影响数据恢复范围），恢复数据时 BR 将同时恢复**系统权限相关数据**。该功能也带来了一些兼容性改变，在默认配置下，`br restore full` 不再支持恢复数据到已有数据的集群。如需恢复数据到已有数据的集群，需要过滤掉相关的系统表，参考命令 `br restore full --filter '.' --filter '!mysql.*'`。
>
```

File: br/br-usage-restore.md [hunk 2]
```diff
@@ -182,22 +183,6 @@ BR 可恢复的**系统权限相关数据**包括如下表：
...
-说明：下表中的 BR v5.4 需要是 >= v5.4.1。
-
-| 恢复版本（横向）\ 备份版本（纵向）   | 用 BR v5.4 恢复 TiDB v5.4 | 用 BR v6.0 恢复 TiDB v6.0 | 用 BR v6.1 恢复 TiDB v6.1| 用 BR v6.2 恢复 TiDB v6.2 |
-|  ----  |  ----  | ---- | ---- | ---- |
-|用 BR v5.4 备份 TiDB v5.4| 兼容 | 不兼容（调整恢复集群的 [新 collation](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 配置跟备份集群相同后，可以恢复）| 不兼容（调整恢复集群的 [新 collation](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 配置跟备份集群相同后，可以恢复） | 不兼容（调整恢复集群的 [新 collation](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 配置跟备份集群相同后，可以恢复）|
-|用 BR v6.0 备份 TiDB v6.0| 不兼容 |兼容 | 兼容 | 兼容 |
-|用 BR v6.1 备份 TiDB v6.1| 不兼容 | 兼容(已知问题，如果备份数据中包含空库可能导致报错，参考 [#36379](https://github.com/pingcap/tidb/issues/36379)) | 兼容 | 兼容 |
-|用 BR v6.2 备份 TiDB v6.2| 不兼容 | 兼容(已知问题，如果备份数据中包含空库可能导致报错，参考 [#36379](https://github.com/pingcap/tidb/issues/36379)) | 兼容 | 兼容 |
-
 自 v6.2.0 起，BR 恢复数据前会检查以下两项：
```
Comment (ran-huang) [inline br/br-usage-restore.md:186]:
```suggestion

自 v6.2.0 起，BR 恢复数据前会检查以下两项：
```

File: br/br-usage-restore.md [hunk 3]
```diff
@@ -183,20 +187,6 @@ BR 可恢复的**系统权限相关数据**包括如下表：
...
-**版本兼容矩阵：**
-
-说明：下表中的 BR v5.4 需要是 >= v5.4.1。
-
-| 恢复版本（横向）\ 备份版本（纵向）   | 用 BR v5.4 恢复 TiDB v5.4 | 用 BR v6.0 恢复 TiDB v6.0 | 用 BR v6.1 恢复 TiDB v6.1| 用 BR v6.2 恢复 TiDB v6.2 |
-|  ----  |  ----  | ---- | ---- | ---- |
-|用 BR v5.4 备份 TiDB v5.4| 兼容 | 不兼容（调整恢复集群的 [新 collation](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 配置跟备份集群相同后，可以恢复）| 不兼容（调整恢复集群的 [新 collation](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 配置跟备份集群相同后，可以恢复） | 不兼容（调整恢复集群的 [新 collation](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 配置跟备份集群相同后，可以恢复）|
-|用 BR v6.0 备份 TiDB v6.0| 不兼容 |兼容 | 兼容 | 兼容 |
-|用 BR v6.1 备份 TiDB v6.1| 不兼容 | 兼容(已知问题，如果备份数据中包含空库可能导致报错，参考 [#36379](https://github.com/pingcap/tidb/issues/36379)) | 兼容 | 兼容 |
-|用 BR v6.2 备份 TiDB v6.2| 不兼容 | 兼容(已知问题，如果备份数据中包含空库可能导致报错，参考 [#36379](https://github.com/pingcap/tidb/issues/36379)) | 兼容 | 兼容 |
```
Comment (IANTHEREAL) [inline br/br-usage-restore.md:190]:
```suggestion
```

File: br/br-usage-restore.md [hunk 4]
```diff
@@ -57,7 +57,11 @@ Full Restore <---------/...............................................> 17.12%.
 
 > **注意：**
 >
-> 自 BR v5.1.0 开始，全量备份会备份**系统表数据**。BR v6.2.0 以前的版本，在默认设置下只会恢复用户数据，而不会恢复系统表数据。自 BR v6.2.0 开始，如果备份数据中包含系统表数据，在默认设置下（即没有 `filter` 等参数影响数据恢复范围），恢复数据时 BR 将同时恢复**系统权限相关数据**。详细内容请查看[恢复 `mysql` 数据库下的表](#恢复-mysql-数据库下的表)
+> 自 BR v5.1.0 开始，全量备份会备份**系统表数据**。BR v6.2.0 以前的版本，在默认设置下恢复只会恢复用户数据，而不会恢复系统表数据。
+> 
+> 自 BR v6.2.0 开始，在默认设置下（即没有 `filter` 等参数影响数据恢复范围），恢复数据时 BR 将同时恢复**系统权限相关数据**。该功能也带来了一些兼容性改变，在默认配置下，`br restore full` 不再支持恢复数据到已有数据的集群。如需恢复数据到已有数据的集群，需要过滤掉相关的系统表，参考命令 `br restore full --filter '.' --filter '!mysql.*'`。
+>
+> 其他详细内容请查看[恢复 `mysql` 数据库下的表](#恢复-mysql-数据库下的表)
```
Comment (ran-huang) [inline br/br-usage-restore.md:64]:
```suggestion
> 其他详细内容请查看[恢复 `mysql` 数据库下的表](#恢复-mysql-数据库下的表)。
```

**Review by ran-huang — APPROVED** [review]

File: br/backup-and-restore-overview.md [hunk 5]
```diff
@@ -81,16 +81,16 @@ BR 和 TiDB 集群的兼容性问题分为两方面：
...
-| 用 BR nightly 或 v5.0 备份 TiDB v4.0 | ❌（当 TiDB 版本小于 v4.0.9 时会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) | ❌（当 TiDB 版本小于 v4.0.9 会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) | ❌（当 TiDB 版本小于 v4.0.9 会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) |
+| 恢复版本（横向）\ 备份版本（纵向）   | 用 BR v5.4 恢复 TiDB v5.4 | 用 BR v6.0 恢复 TiDB v6.0 | 用 BR v6.1 恢复 TiDB v6.1| 用 BR v6.2 恢复 TiDB v6.2 |
+|  ----  |  ----  | ---- | ---- | ---- |
+|用 BR v5.4 备份 TiDB v5.4| 兼容 | 不兼容（调整恢复集群的 [新 collation](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 配置跟备份集群相同后，可以恢复）| 不兼容（调整恢复集群的 [新 collation](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 配置跟备份集群相同后，可以恢复） | 不兼容（调整恢复集群的 [新 collation](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 配置跟备份集群相同后，可以恢复）|
+|用 BR v6.0 备份 TiDB v6.0| 不兼容 |兼容 | 兼容 | 兼容 |
+|用 BR v6.1 备份 TiDB v6.1| 不兼容 | 兼容(已知问题，如果备份数据中包含空库可能导致报错，参考 [#36379](https://github.com/pingcap/tidb/issues/36379)) | 兼容 | 兼容 |
```
Comment (ran-huang) [inline br/backup-and-restore-overview.md:92]:
```suggestion
|用 BR v6.1 备份 TiDB v6.1| 不兼容 | 兼容（已知问题，如果备份数据中包含空库可能导致报错，参考 [#36379](https://github.com/pingcap/tidb/issues/36379)） | 兼容 | 兼容 |
```

File: br/backup-and-restore-overview.md [hunk 6]
```diff
@@ -81,16 +81,16 @@ BR 和 TiDB 集群的兼容性问题分为两方面：
...
+|用 BR v6.2 备份 TiDB v6.2| 不兼容 | 兼容(已知问题，如果备份数据中包含空库可能导致报错，参考 [#36379](https://github.com/pingcap/tidb/issues/36379)) | 兼容 | 兼容 |
```
Comment (ran-huang) [inline br/backup-and-restore-overview.md:93]:
```suggestion
|用 BR v6.2 备份 TiDB v6.2| 不兼容 | 兼容（已知问题，如果备份数据中包含空库可能导致报错，参考 [#36379](https://github.com/pingcap/tidb/issues/36379)） | 兼容 | 兼容 |
```

**Comment by IANTHEREAL** [comment]
this pr should be merged after pr https://github.com/pingcap/tidb/pull/36652/files

**Comment by IANTHEREAL** [comment]
@ran-huang @D3Hunter PTAL

File: br/br-usage-restore.md [hunk 7]
```diff
@@ -166,9 +162,9 @@ br restore full\
 
 ## 恢复 `mysql` 数据库下的表
 
-自 BR v5.1.0 开始，全量备份会备份**系统表数据**。BR v6.2.0 以前的版本，在默认设置下只会恢复用户数据，而不会恢复系统表数据。自 BR v6.2.0 开始，如果备份数据中包含系统表数据，在默认设置下（即没有 `filter` 等参数影响数据恢复范围），恢复数据时 BR 将同时恢复**系统权限相关数据**。
+自 BR v5.1.0 开始，全量备份会备份**系统表数据**。BR v6.2.0 以前的版本，在默认设置下只会恢复用户数据，而不会恢复系统表数据。自 BR v6.2.0 开始，如果备份数据中包含系统表数据，在设置 `--with-sys-table=true` 下，恢复数据时 BR 将同时恢复**部分系统表相关数据**。
```
Comment (D3Hunter) [inline br/br-usage-restore.md:165]:
```suggestion
自 BR v5.1.0 开始，全量备份会备份**系统表数据**。BR v6.2.0 以前的版本，在默认设置下只会恢复用户数据，而不会恢复系统表数据。自 BR v6.2.0 开始，如果备份数据中包含系统表数据，在设置 `--with-sys-table` 下，恢复数据时 BR 将同时恢复**部分系统表相关数据**。
```

**Comment by IANTHEREAL** [comment]
@shichun-0415  I have to tell you, br will not break compatibility after https://github.com/pingcap/tidb/pull/36652

File: br/br-usage-restore.md [hunk 8]
```diff
@@ -183,52 +179,36 @@ BR 可恢复的**系统权限相关数据**包括如下表：
...
-- 目标集群是否为空。
-- 目标集群的系统表是否跟备份数据中的系统表兼容。这里的兼容是指满足以下所有条件：
+- BR 不会恢复 `user` 为 `cloud_admin` 并且 `host` 为 `'%'` 的用户数据，该用户是 TiDB Cloud 预留账户。
+- 在恢复数据前会检查目标集群的系统表是否跟备份数据中的系统表兼容。这里的兼容是指满足以下所有条件:
     - 目标集群需要存在备份中的系统权限表。
     - 目标集群系统权限表**列数**需要跟备份数据中一致，列顺序可以有差异。
     - 目标集群系统权限表列需要跟备份数据兼容，如果为带长度类型（包括整形、字符等类型），前者长度需 >= 后者，如果为 enum 类型，则应该为后者超集。
 
-如果目标集群非空或者目标集群系统表跟备份数据不兼容，BR 会提示类似如下信息。此时可参考提示信息，通过显式指定 `--filter` 的方式跳过恢复系统表:
+如果目标集群非空或者目标集群系统表跟备份数据不兼容，BR 会提示类似如下信息。此时可参考提示信息，通过去掉 `--with-sys-table=true` 配置（或者 `--with-sys-table=false` 参数）的方式跳过恢复系统表:
```
Comment (D3Hunter) [inline br/br-usage-restore.md:196]:
```suggestion
如果目标集群非空或者目标集群系统表跟备份数据不兼容，BR 会提示类似如下信息。此时可参考提示信息，通过去掉 `--with-sys-table` 配置的方式跳过恢复系统表:
```

File: br/br-usage-restore.md [hunk 9]
```diff
@@ -183,52 +179,36 @@ BR 可恢复的**系统权限相关数据**包括如下表：
...
 ```
 
-如需恢复除权限表以外的 `mysql` 数据库下的表，可参考以下方式。
-
-> **警告：**
->
-> 当前该功能为实验特性，不建议在生产环境中使用。
-
-如果需要恢复 `mysql` 下的用户创建的表，可以通过 [table filter](/table-filter.md#表库过滤语法) 来显式地包含目标表。以下示例中要恢复目标用户表为 `mysql.usertable`；该命令会在执行正常的恢复的同时恢复 `mysql.usertable`。
+如果需要恢复 `mysql` 下的用户创建的表（非系统表），可以通过 [table filter](/table-filter.md#表库过滤语法) 来显式地包含目标表。以下示例中要恢复目标用户表为 `mysql.usertable`；该命令会在执行正常的恢复的同时恢复 `mysql.usertable`。
```
Comment (D3Hunter) [inline br/br-usage-restore.md:206]: 同时需要指定 --with-sys-table 参数

Comment (IANTHEREAL) [reply br/br-usage-restore.md:206]: `br restore full -f '*.*' -f '!mysql.*' -f 'mysql.usertable' -s $external_storage_url --with-sys-table`？

还是

`br restore full  -f '!mysql.*' -f 'mysql.usertable' -s $external_storage_url --with-sys-table`？

Comment (IANTHEREAL) [reply br/br-usage-restore.md:206]: https://github.com/pingcap/tidb/pull/36741, 这样子修改是否可以，不过滤掉非系统表

Comment (D3Hunter) [reply br/br-usage-restore.md:206]: 在 mysql 建表属于非常小众的用法吧，参数长点我觉得没啥

这块感觉最好让 tidb 禁止在 mysql 库下建表

File: br/br-usage-restore.md [hunk 10]
```diff
@@ -183,52 +179,36 @@ BR 可恢复的**系统权限相关数据**包括如下表：
...
-|用 BR v6.0 备份 TiDB v6.0| 不兼容 |兼容 | 兼容 | 兼容 |
-|用 BR v6.1 备份 TiDB v6.1| 不兼容 | 兼容(已知问题，如果备份数据中包含空库可能导致报错，参考 [#36379](https://github.com/pingcap/tidb/issues/36379)) | 兼容 | 兼容 |
-|用 BR v6.2 备份 TiDB v6.2| 不兼容 | 兼容(已知问题，如果备份数据中包含空库可能导致报错，参考 [#36379](https://github.com/pingcap/tidb/issues/36379)) | 兼容 | 兼容 |
+当恢复系统权限相关数据的时候，请注意：
 
-自 v6.2.0 起，BR 恢复数据前会检查以下两项：
-
-- 目标集群是否为空。
-- 目标集群的系统表是否跟备份数据中的系统表兼容。这里的兼容是指满足以下所有条件：
+- BR 不会恢复 `user` 为 `cloud_admin` 并且 `host` 为 `'%'` 的用户数据，该用户是 TiDB Cloud 预留账户。
```
Comment (D3Hunter) [inline br/br-usage-restore.md:190]: 这块加个”如果在 OP 环境下备份或恢复集群中有 cloud_admin 用户，可能导致恢复后权限异常，不建议使用” ？

Comment (IANTHEREAL) [reply br/br-usage-restore.md:190]: 这里用了中立的描述，提醒用户不能恢复。 你说的恢复后权限异常，有没有什么特殊的意义，比如除了 cloud admin 不能使用外，有没有其他破坏作用，比如权限表不能工作了之类的

Comment (D3Hunter) [reply br/br-usage-restore.md:190]: 比如备份数据中有个 `cloud_admin`的角色，其他用户绑定到这个角色，因为恢复不完整导致这个用户不能正常使用

Comment (IANTHEREAL) [reply br/br-usage-restore.md]: done

**Comment by IANTHEREAL** [comment]
@D3Hunter @ran-huang PTAL

**Review by D3Hunter — APPROVED** [review]
lgtm

**Review by ran-huang — APPROVED** [review]

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Pull request merged**